pkg/k8s/client/cell.go

```diff
@@ -545,6 +587,86 @@ func FakeClientBuilder() ClientBuilderFunc {
 	}
 }
 
+func FakeClientCommand(fc *FakeClientset) script.Cmd {
+	return script.Command(
+		script.CmdUsage{
+			Summary: "interact with fake k8s client",
+			Args:    "<command> args...",
+		},
+		func(s *script.State, args ...string) (script.WaitFunc, error) {
+			if len(args) < 1 {
+				return nil, fmt.Errorf("usage: k8s <command> files...\n<command> is one of add, update or delete.")
+			}
+
+			action := args[0]
+			if len(args) < 2 {
+				return nil, fmt.Errorf("usage: k8s %s files...", action)
+			}
+
+			for _, file := range args[1:] {
+				b, err := os.ReadFile(s.Path(file))
+				if err != nil {
+					// Try relative to current directory, e.g. to allow reading "testdata/foo.yaml"
+					b, err = os.ReadFile(file)
+				}
+				if err != nil {
+					return nil, fmt.Errorf("failed to read %s: %w", file, err)
+				}
+				obj, gvk, err := testutils.DecodeObjectGVK(b)
+				if err != nil {
+					return nil, fmt.Errorf("decode: %w", err)
+				}
+				gvr, _ := meta.UnsafeGuessKindToResource(*gvk)
+				objMeta, err := meta.Accessor(obj)
+				if err != nil {
+					return nil, fmt.Errorf("accessor: %w", err)
+				}
+				name := objMeta.GetName()
+				ns := objMeta.GetNamespace()
+
+				// Try to add the object to all the trackers. If one of them
+				// accepts we're good. We'll add to all since multiple trackers
+				// may accept (e.g. slim and kubernetes).
+
+				// err will get set to nil if any of the tracker methods succeed.
```

Comment:
We actually don't want to collect the errors as we do expect some of them to fail.

Reply:
Ah actually useful to collect them as the error might be interesting, e.g. you might have 4 trackers rejecting it as unknown object and 1 rejecting it as known but malformed in some way. Changed this to collect all errors.

Reply:
e.g. here's how "double add" looks like now:
```
        scripttest.go:212: FAIL: testdata/hostport.txtar:13: k8s add pod.yaml: none of the trackers of FakeClientset accepted *v1.Pod
            pods "my-app-85f46c4bd9-nnk25" already exists
            no kind is registered for the type v1.Pod in scheme "github.com/cilium/cilium/pkg/k8s/client/clientset/versioned/fake/register.go:18"
            no kind is registered for the type v1.Pod in scheme "sigs.k8s.io/mcs-api/pkg/client/clientset/versioned/fake/register.go:30"
            no kind is registered for the type v1.Pod in scheme "k8s.io/client-go/kubernetes/fake/register.go:82"
            no kind is registered for the type v1.Pod in scheme "k8s.io/apiextensions-apiserver/pkg/client/clientset/clientset/fake/register.go:31"
```

earlier you probably would've just seen "no kind is registered" error and not the "already exists" that you wanted.

Reply:
Thanks! My main point was to use the `success` bool to avoid the need for comments (the current version has even two), but error collection and tracker iteration are also very nice changes (and your solution without bool has the benefit of avoiding the `bool+error` redundancy from my suggestion).